source/extensions/filters/http/cache/cache_header_utility.cc

```diff
@@ -0,0 +1,107 @@
+#include "extensions/filters/http/cache/cache_header_utility.h"
+
+#include <vector>
+
+#include "common/common/utility.h"
+#include "common/http/header_map_impl.h"
+#include "common/http/header_utility.h"
+#include "common/http/headers.h"
+
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/strip.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+namespace {
+const absl::string_view bytes_{"bytes"};
+}
+
+std::vector<RawByteRange> CacheHeaderUtility::getRanges(const Http::HeaderMap& request_headers) {
+  if (request_headers.Method() == nullptr ||
+      request_headers.Method()->value().getStringView() != Http::Headers::get().MethodValues.Get) {
+    // Range headers are only valid on GET requests so don't bother parsing
+    // range if we're not going to use it.
+    return {};
+  }
+
+  // Multiple instances of range/range-unit headers are considered invalid.
+  // https://tools.ietf.org/html/rfc7230#section-3.2.2
+  // TODO: should we log when encountering multiple headers?
+  std::vector<absl::string_view> range_unit_headers;
+  Http::HeaderUtility::getAllOfHeader(request_headers, Http::Headers::get().RangeUnit.get(),
+                                      range_unit_headers);
+  absl::string_view range_unit;
+  if (range_unit_headers.size() == 1) {
+    range_unit = range_unit_headers.front();
+  } else if (range_unit_headers.empty()) {
+    range_unit = bytes_;
```

Comment:
IIUC the spec is a little over-specified in this regard, i.e. I don't think anyone uses anything other than bytes. So, if it's any easier or more efficient to beyond just having it as a default fallback, to be outright required to be specified / the only thing supported, I think you're reasonably free to do that. (Might want a second opinion on that, though).

Reply:
....or should that be "under-specified?" What I actually want to say is "needlessly flexible"

Reply:
I thought about this a while, and you're right that in practice, we'll only ever see `bytes`... I figured the code to support the RFC exactly was trivial enough (and the overhead of reading the header is low enough). If we'd rather not support the overly flexible RFC, then we can remove this bit without any harm done to the overall code.

Reply:
I thought about this a while, and you're right that, in practice, we'll only ever see bytes... I figured the code to support the RFC exactly was trivial enough (and the overhead of reading the header is low enough) to warrant including it. If we'd rather not support the overly flexible RFC, then we can remove this bit without any harm done to the overall code.

Reply:
SGTM

Reply:
We only care whether the range units are "bytes" or not. If somebody adds support for another unit, they can extend the parsing.

Reply:
After fixing up the misunderstanding about `range-unit`, I have it only checking `bytes=`.